File: python-client-sdk/src/sdk/models/shared/codat_assess_datacontracts_common_pagedreportinfo.py
import dataclasses
import dateutil.parser
from dataclasses_json import dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatAssessDataContractsCommonPagedReportInfo:
    company_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('companyName') }})
    generated_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('generatedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    page_number: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageNumber') }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageSize') }})
    report_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('reportName') }})
    total_results: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('totalResults') }})
    

File: go-client-sdk/pkg/models/shared/codatassessdatacontractscommonpagedreportinfo.go
package shared

import (
	"time"
)

type CodatAssessDataContractsCommonPagedReportInfo struct {
	CompanyName   *string    `json:"companyName,omitempty"`
	GeneratedDate *time.Time `json:"generatedDate,omitempty"`
	PageNumber    *int       `json:"pageNumber,omitempty"`
	PageSize      *int       `json:"pageSize,omitempty"`
	ReportName    *string    `json:"reportName,omitempty"`
	TotalResults  *int       `json:"totalResults,omitempty"`
}
